File: cama-tablet/android/app/src/main/java/com/cama/tablet/CamaTabletLog.kt
package com.cama.tablet

import android.util.Log
import com.cama.tablet.BuildConfig

/** 태블릿 앱 통합 로그 — logcat 필터: adb logcat -s CamaTablet:* */
object CamaTabletLog {
    const val TAG = "CamaTablet"

    fun v(message: String) {
        if (BuildConfig.DEBUG) Log.v(TAG, message)
    }

    fun d(message: String) {
        Log.d(TAG, message)
    }

    fun i(message: String) {
        Log.i(TAG, message)
    }

    fun w(message: String, throwable: Throwable? = null) {
        if (throwable != null) Log.w(TAG, message, throwable) else Log.w(TAG, message)
    }

    fun e(message: String, throwable: Throwable? = null) {
        if (throwable != null) Log.e(TAG, message, throwable) else Log.e(TAG, message)
    }

    fun ble(message: String) = i("[BLE] $message")

    fun web(message: String) = i("[WEB] $message")

    fun bridge(message: String) = i("[BRIDGE] $message")
}

File: cama-tablet/ios/CamaTablet/CamaTabletLog.swift
import Foundation
import UIKit
import os.log

enum CamaTabletLog {
    private static let logger = Logger(subsystem: "com.cama.tablet.offline", category: "CamaTablet")

    static func i(_ message: String) {
        logger.info("\(message, privacy: .public)")
    }

    static func w(_ message: String) {
        logger.warning("\(message, privacy: .public)")
    }

    static func e(_ message: String) {
        logger.error("\(message, privacy: .public)")
    }

    static func ble(_ message: String) {
        logger.info("[BLE] \(message, privacy: .public)")
    }

    static func bridge(_ message: String) {
        logger.info("[Bridge] \(message, privacy: .public)")
    }

    static func web(_ message: String) {
        logger.info("[Web] \(message, privacy: .public)")
    }
}
